### DISCUSSION TITLE
Gluten 1.2.1 Discussion

### DISCUSSION THREAD

**Discussion by weiting-chen** [apache/gluten#7125] [General]
This is a discussion to collect all the issues for Gluten v1.2.0 and plan to be fixed in v1.2.1
The concept for version control is 

- Security related issue fixing or Critical bug fixing will be treat as 1st priority to cherry pick into 1.2.1.
- PRs related to new features or spark functions support may require a further discussion. Usually no new features/spark function support in minor version.

Current Velox backend is using gluten-1.2.1(2024-07-05), the related Velox discussion please also put here.

**Comment by wForget** [comment]
Can we port #6679 to 1.2.1? I want to use it in kyuubi, see: https://github.com/apache/kyuubi/issues/6576

**Reply by wForget** [reply]
and #7132

**Reply by weiting-chen** [reply]
#6679 has been ported to branch-1.2 on #7919 
#7132 has been ported to branch-1.2 on #7973

**Comment by weiting-chen** [comment]
Port https://github.com/apache/incubator-gluten/pull/6563 to 1.2.1 based on the request from gluten community thread: https://lists.apache.org/thread/8qqgodr46w3sw9mgxdtxnk92r7o1dvd6

**Reply by weiting-chen** [reply]
#6563 has been ported to branch-1.2 on #7919

**Comment by wForget** [comment]
#7038 fixed the abnormal display of error stackrace when building with vcpkg

**Reply by weiting-chen** [reply]
Done in #7699

**Comment by weiting-chen** [comment]
#7448 for security issue.

**Reply by weiting-chen** [reply]
Done in #7988

**Comment by wForget** [comment]
Gluten 1.2.0 with uniffle bug fixes: #6698 #7559 #7524

**Reply by weiting-chen** [reply]
#6698 has been done in #7994
#7559 #7560 have been done in #7994
#7524 #7525 have been done in #7994

**Comment by ccat3z** [comment]
https://github.com/apache/incubator-gluten/pull/7569 fixed fallback in BNLJ.

**Reply by weiting-chen** [reply]
Done

**Comment by NEUpanning** [comment]
[velox#10755](https://github.com/facebookincubator/velox/pull/10755) fixed result mismatch in decimal aggregation. And it depends on [velox#9868](https://github.com/facebookincubator/velox/pull/9868)

**Reply by NEUpanning** [reply]
[velox#1298](https://github.com/facebookincubator/velox/pull/11298) fixed crash in short decimal aggregation.

**Reply by weiting-chen** [reply]
Done. All 3 PRs have been merged in OAP's Velox branch-1.2.

**Comment by kecookier** [comment]
https://github.com/apache/incubator-gluten/pull/6422  Fix shuffle spill exception.

**Reply by weiting-chen** [reply]
Done

**Comment by ccat3z** [comment]
https://github.com/apache/incubator-gluten/pull/7861 Fix shuffle writer oom

**Reply by weiting-chen** [reply]
Done in #8078
